Shopify/cli
# Dev command in non interactive mode

### WHY are these changes introduced?
As a requirement for running the `dev` command inside a `spin instance` in an automatic way, some modifications should be added to support this feature.



### WHAT is this pull request doing?
- Add new flags to the `dev` command:
  - `--api-secret`: The API secret of your app
  - `--include`: Only the extension which local identifier matches this value will be included in the dev flow execution
  - `--exclude`: All the existing extensions will be included in the dev flow execution except for the one which local identifier matches this value
  - Fetch remote extension specifications skipped in case the command is run in `spin` as a service
  - Add lazy fetching of the extension specifications. Specs are only loaded in case the app includes at least one extension
  - Modified every command that `load` the app to use lazy fetching of the extension specifications so that operation is skipped in case the app has no extensions.

`--include` and `--exclude` flags cannot be used simultaneously. In that case the cli will return an error

![image](https://user-images.githubusercontent.com/105213827/216589571-d08308c1-4c78-49af-93c9-e03c83c4736f.png)

Extensions identifiers used with `--include` and `--exclude` flags should exists. An error is returned in case at least one does not exist:

![image](https://user-images.githubusercontent.com/105213827/216590652-941ef9d5-6809-44b8-a632-836f4411960c.png)



### How to test your changes?
- Write down the `Client ID` and `Client secret` from one of your app using the `Partners dashboard` 

<img src="https://user-images.githubusercontent.com/105213827/210362207-4999bb09-df41-4d21-86a6-90175f358d6f.png" width="500">

- Update app url manually with the ngrok tunnel you are going to use. In spin 1P party apps have fixed app urls so there is no needing on update them that involves prompting the user

- Right now it's not possible to run `dev` command as a service (non-interactive mode) in `spin` for app with extensions, so all of them should be removed from the `dev` workflow using the new `--exclude` flag. 
This is an example of how to run the `dev` command in a non-interactive mode with two extensions 

```
SPIN=1 SHOPIFY_FLAG_APP_API_KEY=8289684b401bbac67177ec6728d120b3 SHOPIFY_FLAG_APP_API_SECRET=c6fd90b7d741d8c331eb0b1ce355b615 SHOPIFY_FLAG_STORE=cli-second.myshopify.com pnpm shopify app dev --path ~/cli-apps/last-node2 --no-update --tunnel-url https://9b50-47-62-197-187.ngrok.io:5675 --exclude checkout-02 --exclude theme-app-extension 
```

### Post-release steps


- Modify the documentation to add the new parameter
- Add new instrumentation information about the usage of the `non-interactive dev mode`

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
